src/core/build_simulations.jl

```diff
@@ -80,17 +80,29 @@ function _build_stages(sim_ref::SimulationRef,
 
     for (k, v) in stages
```

Comment:
If "v" is a Stage then can you call it "stage" instead?  It would be more clear.